packages/makeswift/src/init.ts

```diff
@@ -33,85 +33,103 @@ export default async function wrappedInit(name: string | undefined, args: InitAr
+    const nextAppUrl = `http://localhost:${nextAppPort}`
+    const redirectUrl = new URL(`${MAKESWIFT_APP_ORIGIN}/cli/link-site`)
+    redirectUrl.searchParams.set('host_url', nextAppUrl)
+
+    // Handshake Step 3 - we redirect the browser to redirectUrl
+    const { siteApiKey, example } = await getSiteMetadata({
+      port: handshakePort,
+      redirectUrl: redirectUrl.toString(),
+    })
+
+    // In the background, we're setting up the Next app with the API key
+    // and starting the app at `nextAppPort`
+    const envLocal = buildLocalEnvFile(
+      {
+        MAKESWIFT_SITE_API_KEY: siteApiKey,
+        MAKESWIFT_API_ORIGIN,
+      },
+      env,
     )
+
+    // Handshake Step 4 - Makeswift redirects to the builder with the site open,
+    //                    with the host using `nextAppUrl` for the builder
+
+    return { nextAppPort, envLocal, example }
   }
 
+  validate()
+
   const { directory: nextAppDir, name: projectName } = await getProjectName(name)
+  const usingExistingNextApp = isNextApp(nextAppDir)
+
+  checkForConflictingFiles({ dir: nextAppDir })
 
-  if (isNextApp(nextAppDir)) {
-    checkForConflictingFiles({ dir: nextAppDir })
+  const { nextAppPort, envLocal, example } = await performHandshake({ usingExistingNextApp })
 
+  if (usingExistingNextApp) {
     await integrateNextApp({ dir: nextAppDir })
   } else {
     createNextApp({
       dir: nextAppDir,
-      example,
+      example: example || 'basic-typescript',
       useNpm,
       usePnpm,
     })
   }
 
-  const handshakePort = await detect(5600)
-  const nextAppPort = await detect(3000)
-
-  const callbackUrl = `http://localhost:${handshakePort}/${siteSelectionPath}`
-  // Handshake Step 1
-  const selectSiteUrl = new URL(`${MAKESWIFT_APP_ORIGIN}/cli/select-site`)
-  selectSiteUrl.searchParams.set('project_name', projectName)
-  selectSiteUrl.searchParams.set('callback_url', callbackUrl)
-
-  const selectSiteUrlString = selectSiteUrl.toString()
-  console.log(
-    `\nOpening your browser at ${chalk.blue(
-      selectSiteUrlString,
-    )}\n\nIf anything happens, re-open that URL.`,
-  )
-  await open(selectSiteUrlString)
-
-  // Handshake Step 2 - the browser goes to `callbackUrl`
-  const nextAppUrl = `http://localhost:${nextAppPort}`
-  const redirectUrl = new URL(`${MAKESWIFT_APP_ORIGIN}/cli/link-site`)
-  redirectUrl.searchParams.set('host_url', nextAppUrl)
-
-  // Handshake Step 3 - we redirect the browser to redirectUrl
-  const { siteApiKey } = await getSiteApiKey({
-    port: handshakePort,
-    redirectUrl: redirectUrl.toString(),
-  })
-
-  // In the background, we're setting up the Next app with the API key
-  // and starting the app at `nextAppPort`
-  const envLocal = buildLocalEnvFile(
-    {
-      MAKESWIFT_SITE_API_KEY: siteApiKey,
-      MAKESWIFT_API_ORIGIN,
-    },
-    env,
-  )
   fs.writeFileSync(`${nextAppDir}/.env.local`, envLocal)
 
   spawn.sync('yarn', ['dev', '--port', nextAppPort.toString()], {
```

Comment:
What happens here is they don't have Yarn installed and are using NPM?

Reply:
Good catch. This could be what Lindsay was hitting. I'll address this in another PR